src/main/java/alien4cloud/paas/yorc/context/service/fsm/FsmBuilder.java

```diff
@@ -65,6 +65,49 @@ private void configure(StateMachineTransitionConfigurer<FsmStates, FsmEvents> tr
 			.withExternal()
 			.source(FsmStates.DEPLOYMENT_IN_PROGRESS).target(FsmStates.FAILED)
 			.event(FsmEvents.FAILURE)
+
+// Update stuff here
+// Allow update from DEPLOYED state
+		.and()
+			.withExternal()
+			.source(FsmStates.DEPLOYED).target(FsmStates.UPDATE_IN_PROGRESS)
+			.event(FsmEvents.UPDATE_STARTED)
+			.action(actions.update())
+// Allow update from UPDATED state
+		.and()
+			.withExternal()
+			.source(FsmStates.UPDATED).target(FsmStates.UPDATE_IN_PROGRESS)
+			.event(FsmEvents.UPDATE_STARTED)
+			.action(actions.update())
+// Allow update from UPDATE_FAILED state
+		.and()
+			.withExternal()
+			.source(FsmStates.UPDATE_FAILED).target(FsmStates.UPDATE_IN_PROGRESS)
+			.event(FsmEvents.UPDATE_STARTED)
+			.action(actions.update())
+// Success
+		.and()
+			.withExternal()
+			.source(FsmStates.UPDATE_IN_PROGRESS).target(FsmStates.UPDATED)
+			.event(FsmEvents.UPDATE_SUCCESS)
+// Failure
+		.and()
+			.withExternal()
+			.source(FsmStates.UPDATE_IN_PROGRESS).target(FsmStates.UPDATE_FAILED)
+			.event(FsmEvents.FAILURE)
+// Allow undeploy from UPDATED state
+		.and()
+			.withExternal()
+			.source(FsmStates.UPDATED).target(FsmStates.UNDEPLOYMENT_IN_PROGRESS)
+			.event(FsmEvents.UNDEPLOYMENT_STARTED)
+			.action(actions.undeploy())
+// Allow undeploy from UPDATE_FAILED state
+		.and()
+			.withExternal()
+			.source(FsmStates.UPDATE_FAILED).target(FsmStates.UNDEPLOYMENT_IN_PROGRESS)
+			.event(FsmEvents.UNDEPLOYMENT_STARTED)
+			.action(actions.undeploy())
+
```

Comment:
Missing transitions on EVICTION event

Reply:
Can you please give more details ?

Reply:
We need to add Eviction transitions by inserting this:

```suggestion
		.and()
			.withExternal()
			.source(FsmStates.UPDATE_FAILED).target(FsmStates.UNDEPLOYED)
			.event(FsmEvents.EVICTION)
			.action(actions.cleanup())
		.and()
			.withExternal()
			.source(FsmStates.UPDATED).target(FsmStates.UNDEPLOYED)
			.event(FsmEvents.EVICTION)
			.action(actions.cleanup())
		.and()
			.withExternal()
			.source(FsmStates.UPDATE_IN_PROGRESS).target(FsmStates.UNDEPLOYED)
			.event(FsmEvents.EVICTION)
			.action(actions.cleanup())			
```

When a deployment is alive (in a state other than UNDEPLOYED) in a4c but does not exist any more in yorc, the eviction thread use theses transitions to set the deployment as UNDEPLOYED

Reply:
OK. Fixed.